lite/src/skipping.rs

```diff
@@ -0,0 +1,94 @@
+use crate::types::*;
+
+/// SkippingVerifier tries to skip headers by verifying if
+/// +1/3 of the last validators it trusts has signed the new header.
+struct SkippingVerifier<H, V>
+where
+    H: Header,
+    V: ValidatorsLookup,
+{
+    trusting_period: Time,
+    state: TrustedState<H, V>,
+}
+
+impl<H, V> SkippingVerifier<H, V>
+where
+    H: Header,
+    V: ValidatorsLookup,
+{
+    /// Returns the time after which the current state expires
+    /// and the verifier must be reset subjectively.
+    fn expires(&self) -> Time {
+        self.state.header.bft_time() + self.trusting_period
+    }
+
+    /// Verify takes a header, a commit for the header,
+    /// and the next validator set referenced by the header.
+    /// Note we do not know the correct validator set for this commit -
+    /// we can only check if it was signed by enough of the validators we do know about.
+    /// Returns an error if verification fails.
+    fn verify<C>(self, now: Time, header: H, commit: C, next_validators: V) -> Result<(), Error>
+    where
+        C: Commit,
+    {
+        // ensure the state is not expired.
+        if self.expires() < now {
+            return Err(Error::Expired);
+        }
+
```

Comment:
Should we have check that `header.Height > self.state.header.height() + 1`, i.e., that we are not in sequential case?